saylua/modules/pets/views.py

```diff
@@ -1,18 +1,83 @@
-from flask import render_template, redirect
+from flask import render_template, g, redirect, request, flash
+from saylua import db
+from .models.db import Pet
 
 
 def pet_profile(name):
-    return render_template("profile.html")
+    if request.method == 'POST':
+        if not g.user:
+            flash("You need to be logged in to do this!", "error")
+            return redirect("/login")
+        companion_id = request.form.get('accompany')
```

Comment:
Personally, I feel like putting both abandoning and making pets active here is confusing and overloaded. I'd split these into two separate routes and make both redirect to the profile.